include/mitsuba/core/transform.h

```diff
@@ -52,45 +60,66 @@ template <typename Point_> struct Transform {
     //! @{ \name Constructors, methods, etc.
     // =============================================================
 
-    /// Initialize the transformation from the given matrix (and compute its inverse transpose)
-    Transform(const Matrix &value)
-        : matrix(value),
-          inverse_transpose(dr::inverse_transpose(value)) { }
+    /// Default constructor - identity transformation
+    Transform() = default;
+
+    /// Initialize the transformation from the given matrix
+    Transform(const Matrix &value) {
+        matrix = value;
+        update(); // Compute the inverse transpose
+    }
 
     /// Initialize the transformation from the given matrix and its inverse
-    Transform(const Matrix &value, const Matrix &inv)
-        : matrix(value),
-          inverse_transpose(inv) { }
+    Transform(const Matrix &matrix, const Matrix &inverse_transpose)
+        : matrix(matrix), inverse_transpose(inverse_transpose) { }
 
-    template <typename T,  dr::enable_if_t<!std::is_same_v<T, Point_>> = 0>
-    Transform(const Transform<T> &transform)
-        : matrix(dr::Matrix<dr::scalar_t<T>, Size>(transform.matrix)),
-          inverse_transpose(dr::Matrix<dr::scalar_t<T>, Size>(transform.inverse_transpose)) { }
+    /// Copy constructor with type conversion
+    template <typename OtherPoint, bool OtherAffine>
+    Transform(const Transform<OtherPoint, OtherAffine> &other)
+        : matrix(Matrix(other.matrix)),
+          inverse_transpose(Matrix(other.inverse_transpose)) { }
 
-    /// Concatenate transformations
-    MI_INLINE Transform operator*(const Transform &other) const {
-        return Transform(matrix * other.matrix,
-                         inverse_transpose * other.inverse_transpose);
-    }
-
-    /// Compute the inverse of this transformation (involves just shuffles, no arithmetic)
     MI_INLINE Transform inverse() const {
-        return Transform(dr::transpose(inverse_transpose), dr::transpose(matrix));
+        return Transform(dr::transpose(inverse_transpose),
+                         dr::transpose(matrix));
```

Comment:
In all classes where we had stored both directions (e.g. `m_camera_to_sample`, `m_sample_to_camera`), are these additional 2 `dr::transpose()` calls every time the field is used worth it?

Reply:
For the JIT, those don't matter. For scalar mode, there might be a cost (but I would expect this to be rather negligible).